docs/src/tutorial/clibraries.rst

```diff
@@ -232,10 +252,56 @@ Here is the most basic script for compiling a Cython module::
         ext_modules = cythonize([Extension("queue", ["queue.pyx"])])
     )
 
-To build against the external C library, we must extend this script to
-include the necessary setup.  Assuming the library is installed in the
-usual places (e.g. under ``/usr/lib`` and ``/usr/include`` on a
-Unix-like system), we could simply change the extension setup from
+
+To build against the external C library, we need to make sure Cython finds the necessary libraries. 
+There are two ways to archive this. Firstly we can tell distutils where to find the c-source to compile the queue.c implementation automatically. Alternatively we can build and install C-Alg as system library and dynamically link it. Latter is useful if other applications also use C-Alg.
+
+
+Static Linking
+---------------
+
+To build the c-code automatically we need to include compiler directives in `queue.pyx`::
+
+    # file: queue.pyx
+
+    # distutils: language = c
+    # distutils: sources = c-algorithms/src/queue.c
+    # distutils: include_dirs = c-algorithms/src/ 
+
+    cimport cqueue
+
+    cdef class Queue:
+        cdef cqueue.Queue* _c_queue
+        def __cinit__(self):
+            self._c_queue = cqueue.queue_new()
+            if self._c_queue is NULL:
+                raise MemoryError()
+
+
+        def __dealloc__(self):
+            if self._c_queue is not NULL:
+                cqueue.queue_free(self._c_queue)
+
+
+In general all relevant header files should be found in ``include_dirs``.
+Now we can build the project using::
+
+    $ python setup.py build_ext -i 
+
+And test whether our build was successful::
+
+    $ python -c 'import queue; Q = queue.Queue()' 
+
+
+Dynamic Linking
+---------------
+
+Dynamic Linking is useful, if the library we are going to wrap is already installed on the system. 
+To perform dynamic linking we first need to build and install c-alg. Follow the instruction in paragraph 
+"Download & Install [CAlg]" to install the library. Afterwards the file `/usr/local/lib/libcalg.so` should exist. 
```

Comment:
Better use a real ReST reference to the section here.
"/usr/local/lib/libcalg.so" is obviously platform specific. Better add a note about it.